SND/MTC/MTCDetector.cxx

```diff
@@ -496,13 +494,28 @@ void MTCDetector::SiPMOverlap()
     }
     Double_t fLengthScifiMat = fSciFiActiveY;
     Double_t fWidthChannel = fFiberPitch * fChannelAggregated;
-    Int_t fNSiPMChan = fNumberofChannels;
-    Int_t fNSiPMs = std::ceil(fWidth / (fNSiPMChan * fWidthChannel));
-    // define the edge shift. In case of N SiPMs, in cm. Example: for 8 SiPMs with 2 channel aggregation,
-    // fEdge = -1.2/2 cm
-    Double_t fEdge = -(fWidth - fNSiPMs * fNSiPMChan * fWidthChannel) / 2;
+    fNSiPMChan = std::ceil(fWidth / fWidthChannel);
+    if (fNSiPMChan > 1000) {
+        LOG(info) << "Number of SiPM channels exceed 1000, namely: " << fNSiPMChan;
```

Comment:
Is the limitation of 1000 due to the channel numbering?

Maybe we can do something like:
```cpp
  // Maximum channels per SiPM limited by 3-digit encoding in detector ID
  static constexpr Int_t kMaxChannelsPerSiPM = 1000;

  if (fNSiPMChan > kMaxChannelsPerSiPM) {
      LOG(warn) << "Number of SiPM channels (" << fNSiPMChan
                << ") exceeds maximum per SiPM (" << kMaxChannelsPerSiPM
                << "), redistributing across multiple SiPMs";
```